### PR TITLE
Bugfix/771 test error beforetest

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

**Comment by sksamuel** [comment]
I created a PR so I could review it on here.

File: kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt [hunk 1]
```diff
@@ -193,9 +187,11 @@ class JUnitTestRunnerListener(private val listener: EngineExecutionListener,
       logger.error("Spec descriptor cannot be null $description")
       throw RuntimeException("Spec descriptor cannot be null")
     } else {
-      val result = when (t) {
-        null -> TestExecutionResult.successful()
-        else -> TestExecutionResult.failed(t)
+      val firstTestError = results.find { it.result.error != null }?.result?.error
```
Comment (sksamuel) [inline kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt]: I am pretty sure that the JUnit specs say that a parent must pass even if it's children fail.
This is why I do not search the children for a failed test.

File: kotlintest-runner/kotlintest-runner-console/src/main/kotlin/io/kotlintest/runner/console/TeamCityConsoleWriter.kt [hunk 2]
```diff
@@ -1,17 +1,14 @@
...
+import io.kotlintest.*
 import java.io.PrintWriter
 import java.io.StringWriter
 import kotlin.reflect.KClass
 
 class TeamCityConsoleWriter : ConsoleWriter {
 
+  private val results = HashSet<TestResult>()
+
   private var errors = false
```
Comment (sksamuel) [inline kotlintest-runner/kotlintest-runner-console/src/main/kotlin/io/kotlintest/runner/console/TeamCityConsoleWriter.kt]: Is this read anywhere?

File: kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt (hunk 1 above)
Comment (LeoColman) [reply kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt]: Perhaps. Let me see what I can do.

Comment (LeoColman) [reply kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt]: The thing is, if we don't fail the parent, the test case won't ever be created

Comment (LeoColman) [reply kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt]: and it won't show in IntelliJ, that's why this fix was necessary

Comment (LeoColman) [reply kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt]: perhaps we can add the dummy failure like we do with our runner

**Pull request closed without merging**